integration-cli/docker_cli_exec_test.go

```diff
@@ -595,3 +595,13 @@ func (s *DockerSuite) TestExecOnReadonlyContainer(c *check.C) {
 		c.Fatalf("exec into a read-only container failed with exit status %d", status)
 	}
 }
+
+// #15750
+func (s *DockerSuite) TestExecStartFails(c *check.C) {
+	name := "exec-15750"
+	dockerCmd(c, "run", "-d", "-t", "--name", name, "busybox", "/bin/sh")
```

Comment:
@duglin  this only tests the else branch(https://github.com/mountkin/docker/blob/fix-exec-panic/api/server/exec.go#L98). 
If I add the `-t -i` arguments docker client complaints that it's not a tty. To test the `if !execStartCheck.Detach {` branch I think we should call the exec APIs directly. Do you think I should do this?